server/utils/verifiers/VerifierBase.ts

```diff
@@ -0,0 +1,350 @@
+                        result.edan = details;
+                        result.details?.push(details);
+                        break;
+                    }
+                }
+            } else {
+                // TODO: create new identifier, type, and of EDAN Record ID type
+            }
+        }
+
+        // see if we have an ARK id stored
+        if(record.identifierPublic) {
+            // get the identifier if it exists
+            const arkIdentifiers: DBAPI.Identifier[] | null = await DBAPI.Identifier.fetchFromIdentifierValue(record.identifierPublic);
+            if(arkIdentifiers) {
+                // cycle through and get our type and details
+                for(const identifier of arkIdentifiers) {
+                    const details: IdentifierDetails | null = await this.getIdentifierType(identifier);
+                    if(!details) {
+                        LOG.error(`could not get details for ARK identifier (type: ${identifier.idVIdentifierType} | value:${identifier.IdentifierValue})`, LOG.LS.eTEST);
+                        continue;
+                    }
+
+                    // if we have an identifier that is of the same type then store
+                    // todo: verify it's an actual ARK id
+                    if(details.identifierTypeEnum===COMMON.eVocabularyID.eIdentifierIdentifierTypeARK) {
+                        result.ark = details;
+                        result.details?.push(details);
+                        break;
+                    }
+                }
+            } else {
+                // todo: create new identifier, type, and of ARK type
+            }
+        }
+
+        // handle identifiers by checking if any returned by EDAN
+        if(record.identifierMap) {
+            for (const [ label, content ] of record.identifierMap) {
+
+                // get our type for this identifier
+                const identifierType: DBAPI.Vocabulary | undefined = await CACHE.VocabularyCache.mapIdentifierType(label);
+                if (!identifierType) {
+                    LOG.error(`\tencountered unknown identifier type ${label} for EDAN record ${record.name}`, LOG.LS.eTEST);
+                    continue;
+                }
+
+                // pull from enumeration from the CACHE (vocabulary id -> enum)
+                const identifierTypeEnum: COMMON.eVocabularyID | undefined = await CACHE.VocabularyCache.vocabularyIdToEnum(identifierType.idVocabulary);
+                if(identifierTypeEnum===undefined){
+                    LOG.error(`\tcould not find enumerator for identifier type (${identifierType.Term}) in Cache`, LOG.LS.eTEST);
+                    continue;
+                }
+
+                // if identifier exists in our database (value & type) then store it
+                const identifiers: DBAPI.Identifier[] | null = await DBAPI.Identifier.fetchFromIdentifierValue(content);

```

Comment:
I'm confused by the intention of this code. `fetchFromIdentifierValue` is going to get *all* identifiers that match the specified value ... this may include matches for identifiers with the same value, but applied to other system objects, and/or of random identifier types.  In other words, identifiers are not guaranteed to have a unique value....